examples/compare_connectivity_over_time_over_trial.py

```diff
@@ -0,0 +1,392 @@
+#
+# There are multiple different spectral connectivity measures, e.g. coherence,
+# imaginary part of coherence, phase locking value, and envelope correlations.
+# Additionally, there are also multiple methods to estimate the frequency
+# content, e.g. Fourier transforms, Morlet wavelets and multitapers.
+#
+# In this example, we focus on the two functions
+# :func:`mne_connectivity.spectral_connectivity_epochs` and
+# :func:`mne_connectivity.spectral_connectivity_time`,
+# which both computes connectivity from epoched data :class:`mne.Epochs`.
+#
+# Both functions contain the options to choose the connectivity measurements
+# of interest and the method to decompose the frequency content.
+# The crucial difference between when to use which function lies on
+# the experimental setup and type of tasks.
+#
+# If the data is obtained for repeated trials of a time-locked event,
+# e.g. ERP data, then :func:`mne_connectivity.spectral_connectivity_epochs`
+# is most likely the corresponding function to utilize.
+#
+# If the data is obtained from resting-state or free-moving tasks,
+# e.g. a mirror-game paradigm, then
+# :func:`mne_connectivity.spectral_connectivity_time` might be better suited.
+#
+# The way connectivity is computed for the two functions are slightly different,
+# thus their interpretations and the hypotheses being tested are also different.
+#
+# Connectivity over trials, as computed by
+# :func:`mne_connectivity.spectral_connectivity_epochs` assume epochs have
+# been created around each trial and the function estimates the connectivity
+# at each timepoint over all the repeated trials for the same event.
+# This gives a high temporal resolution, which is often desired for ERP
+# analysis where you are interested in when an effect occurs.
+# However, this approach is not feasible for single-trial data and will
+# result in errorful values when computed on one epoch.
+#
+# On the other hand, if you are interested in determining connectivity over
+# time for single-trials or from experiments that do not involve exactly
+# repeated trials, then :func:`mne_connectivity.spectral_connectivity_time`
+# should be employed. This function also takes data in the form of
+# :class:`mne.Epochs`, but it may consist of a single epoch.
+# If there are multiple epochs, the connectivity over time is computed for
+# epoch separately, with the option to average over epochs.
+#
+# To better illustrate the differences between the two functions,
+# we will employ them on two simulated cases.
+
+###############################################################################
+# Simulated examples
+# ------------------
+# **Case 1: Repetitions of the same trial.**
+#
+# Let's generate some simulated data in the format of :class:`mne.EpochsArray`.
+# In this case, we will use random data for 3 channels over 5 epochs, but
+# all the epochs are just exact replicates of the first epoch.
+
+np.random.seed(1234) # set seed for reproducibility
```

Comment:
I would set this at the very top. And instead of setting a seed, I would use a random number generator, which is considered more stable (https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html).

You can do this with `rng = np.default_rng(1234)` and then everytime you have `np.random.*`, you replace it with `rng.*` 

See: https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.default_rng